plugins/modules/host_template.py

```diff
@@ -0,0 +1,263 @@
+        try:
+            self.host_template = host_temp_api_instance.read_host_template(
+                cluster_name=self.cluster_name,
+                host_template_name=self.name,
+            )
+        except ApiException as ex:
+            if ex.status == 404:
+                pass
+            else:
+                raise ex
+
+        if self.host_template:
+            if self.module._diff:
+                current = {
+                    item.role_config_group_name
+                    for item in self.host_template.role_config_group_refs
+                }
+                incoming = set(self.role_groups)
+                self.diff.update(
+                    before=list(current - incoming), after=list(incoming - current)
+                )
+
+        if self.state == "present":
+            host_template_body = ApiHostTemplate(
+                cluster_ref=ApiClusterRef(
+                    cluster_name=self.cluster_name, display_name=self.cluster_name
+                ),
+                name=self.name,
+                role_config_group_refs=[
+                    ApiRoleConfigGroupRef(role_config_group_name=group)
+                    for group in self.role_groups
+                ],
+            )
+            if self.host_template:
+                if not self.module.check_mode:
+                    host_temp_api_instance.update_host_template(
+                        cluster_name=self.cluster_name,
+                        host_template_name=self.name,
+                        body=host_template_body,
+                    )
+                    self.changed = True
+            else:
+                body = ApiHostTemplateList(items=[host_template_body])
+                if not self.module.check_mode:
+                    host_temp_api_instance.create_host_templates(
+                        cluster_name=self.cluster_name, body=body
+                    )
+                    self.changed = True
+
+            self.host_template_output = _parse_host_template_output(
+                host_temp_api_instance.read_host_template(
+                    cluster_name=self.cluster_name,
+                    host_template_name=self.name,
+                ).to_dict()
+            )
+
+        if self.state == "absent":
+            if not self.module.check_mode:
+                self.host_template_output = _parse_host_template_output(
+                    host_temp_api_instance.delete_host_template(
+                        cluster_name=self.cluster_name,
+                        host_template_name=self.name,
+                    ).to_dict()
+                )
+                self.changed = True
+
+
+def main():
+    module = ClouderaManagerModule.ansible_module(
+        argument_spec=dict(
+            cluster=dict(required=True, type="str", aliases=["cluster_name"]),
+            name=dict(required=True, type="str"),
+            role_groups=dict(
+                required=True, type="list", aliases=["role_config_groups"]
```

Comment:
Still is showing `required`. This needs to moved to a `required_if` that is pinned to `state=present` condition.